ex02.exs

```diff
@@ -81,24 +81,21 @@ end
 defmodule Ex02 do
 
   def new_counter(initial_value \\ 0) do
-    # « your code goes here»
+    Agent.start_link(fn -> initial_value end, name: __MODULE__)
   end
 
   def next_value(agent) do
-    # « your code goes here»
+    Agent.get_and_update(__MODULE__, fn state -> {state, state+1} end )
+    #Agent.update(__MODULE__, &(&1+1))
```

Comment:
I always tell people that warnings should be treated as errors, and this is a good example of that.

The intent of the API (shown in the tests) is that in this first set of functions the `new_counter` function returns a pid which is then passed to `next_value`. The warning that `agent` is unused it a hint that something's not right.

# 0